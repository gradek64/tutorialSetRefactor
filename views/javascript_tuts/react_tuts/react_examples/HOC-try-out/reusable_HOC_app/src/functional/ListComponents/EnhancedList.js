import React from 'react';
import PropTypes from 'prop-types';
import PureList from './DisplayList';
import { manipulateListFns } from './manipulateListData';

/*
  *@this Component is resposible of extending baseList component but
  it will transform/change props according to the actionType given from the parent Component
  as filering,sorting etc..
*/

/*
  *@There is 2 wasy of making your EnhancedCompoent (component that changes is baseComponent)
  *@1) wrap in a function that passes the baseCompoent into Enhanced compoent and return BaseCompoent
  *@1) use  baseCompoent directly as return value as below
*/

// const withTransformProps = helperFunc => BaseComponent => baseProps => {
const withTransformProps = (helperFunc) => {
  const withTransformPropsComp = (props) => {
    /** ** props is a object defined in propTypes
    * */
    console.log('props of ListWithTransformation:', props);

    /*
      pure list takes only {list:[]} but we do not validate that here;
      It is validated in <PureList list=[] />
    */
    const transformedProps = helperFunc(props);
    return <PureList {...transformedProps} />;
  };

  withTransformPropsComp.propTypes = {
    list: PropTypes.instanceOf(Array).isRequired,
    letterSearch: PropTypes.string,
    sortBy: PropTypes.string,
    filter: PropTypes.string,
    actionType: PropTypes.string,
  };

  withTransformPropsComp.defaultProps = {
    letterSearch: '',
    sortBy: '',
    filter: '',
    actionType: '',
  };

  return withTransformPropsComp;
};
// const ListWithTransformation = withTransformProps(manipulateListFns)(RenderDisplayList);
const ListWithTransformation = withTransformProps(manipulateListFns);

export default ListWithTransformation;
