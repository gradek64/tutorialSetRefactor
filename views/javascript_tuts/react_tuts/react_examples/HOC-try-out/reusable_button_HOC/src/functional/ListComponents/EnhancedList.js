import React from "react";
import PureList from "./DisplayList";
import { manipulateListFns } from "./manipulateListData";

// const withTransformProps = helperFunc => BaseComponent => baseProps => {
const withTransformProps = helperFunc => props => {
  const transformedProps = helperFunc(props);
  return <PureList {...transformedProps} />;
};
// const ListWithTransformation = withTransformProps(manipulateListFns)(RenderDisplayList);
const ListWithTransformation = withTransformProps(manipulateListFns);

export default ListWithTransformation;
