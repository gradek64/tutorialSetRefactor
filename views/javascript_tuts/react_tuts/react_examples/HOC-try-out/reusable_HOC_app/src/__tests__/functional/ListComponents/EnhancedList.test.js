import React from 'react';
import { shallow } from 'enzyme';

// component for snapshot test;
import EnhancedList from '../../../functional/ListComponents/EnhancedList';

let wrapper;
beforeAll(() => {
  wrapper = shallow(<EnhancedList list={[
    { name: 'Gregi', side: 'noSide' },
    { name: 'Greg', side: 'dark' },
  ]}
  />);
});

test('renders correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('tranformProps gives back a correct Object', () => {
  // property name exist in the list
  expect(wrapper.prop('list')[0].name).toBeDefined();
  // property side exist in the list
  expect(wrapper.prop('list')[0].side).toBeDefined();
});
