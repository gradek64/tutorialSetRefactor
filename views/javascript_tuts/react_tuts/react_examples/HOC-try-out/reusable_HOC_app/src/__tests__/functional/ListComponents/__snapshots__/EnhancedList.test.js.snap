// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RenderDisplayList
  list={
    Array [
      Object {
        "name": "Gregi",
        "side": "noSide",
      },
      Object {
        "name": "Greg",
        "side": "dark",
      },
    ]
  }
/>
`;
